# REIT STOCK ANALYSIS

In this project, you will analyze Real Estate Investment Trusts, commonly known as REITs. REITs are companies that own or operate real estate that produces income. REITs, like the stocks of regular public companies, are traded on different stock exchanges. Investing in a REIT allows you to invest in portfolios of real estate assets the same way you can invest in a company by buying its stock.

Using financial statistics and NumPy you will analyze two REITs: [Sabra Health Care REIT Inc. (NASDAQ: SBRA)](https://finance.yahoo.com/quote/SBRA/), which invests in health care real estate, and [Equity Residential (NASDAQ:EQR)](https://finance.yahoo.com/quote/EQR/), which invests in rental apartment properties.

The time period for analysis we will be using is `Jan 1 2018` to `Dec 31 2018`. The REIT data for SBRA (`SBRA.csv`) and EQR (`EQR.csv`) can be found in the same folder as this file.

1. Import the numpy module as np

In [3]:
import numpy as np

2. Load the adjusted closings for SBRA

In [17]:
SBRA = np.loadtxt("SBRA.csv",delimiter=",",skiprows=1,usecols=5)

3. Load the adjusted closings for EQR

In [15]:
EQR = np.loadtxt("EQR.csv",delimiter=",",usecols=5, skiprows=1)
print(EQR)

[62.800674 63.19466  63.588642 64.120514 65.322166 65.322166 66.395767
 66.533653 66.61245  67.429955 68.040634 68.858147 69.064987 69.340775
 69.28167  69.291527 69.606705 70.640915 71.281136 70.916695 71.468269
 70.936401 71.694817 71.852402 71.901649 72.433533 72.335037 72.354729
 71.448578 72.187286 72.019852 72.10849  72.423676 71.586472 72.748718
 73.113152 72.394127 72.610817 72.226692 72.581276 72.394127 72.866913
 72.817657 72.443382 72.325188 72.532021 73.319992 73.920807 73.782921
 73.930656 73.635178 72.866913 72.394127 72.512329 73.664719 73.841385
 73.910858 74.575829 74.41703  75.062149 74.754478 74.41703  75.121704
 74.823952 74.714775 75.042297 74.992676 74.823952 75.776741 76.114189
 76.868484 77.017357 75.1614   73.732208 74.506355 73.613106 74.397179
 74.903351 75.399597 75.627869 74.734627 75.846214 75.945465 75.82637
 76.014938 75.637794 74.208603 73.881088 74.258232 75.260651 75.270569
 75.240791 75.61795  76.233284 76.044716 75.181252 75.786667 76.064568
 76.531

## Simple Rate of Return Function

4. To calculate the daily rate of return for the SBRA stock, we need the daily adjusted closing price. The formula we are using for the daily rate of return is out[n] = a[n+1] - a[n] 

5. Create a function that returns the daily rate of return

    step 1. define a function named `rate_of_return`
    
    step 2. create parameter for  `adj_closings`
    
    step 3. Within the function use np.diff() and set it to the variable `daily_simple_ror`
    
    step 4. return `daily_simple_ror`

In [37]:
def rate_of_return(adj_closing):
    daily_simple_ror = np.diff(adj_closing)/adj_closing[:-1]
    return daily_simple_ror
print(rate_of_return(EQR))
srorEQR= rate_of_return(EQR)

[ 0.0062736   0.00623442  0.00836426  0.01874052  0.          0.01643548
  0.00207673  0.00118432  0.01227256  0.00905649  0.01201507  0.00300386
  0.00399317 -0.00085238  0.00014227  0.00454858  0.01485791  0.00906303
 -0.00511273  0.00777777 -0.00744202  0.01069149  0.002198    0.00068539
  0.00739738 -0.00135981  0.00027223 -0.01252373  0.01033902 -0.00231944
  0.00123074  0.004371   -0.01155981  0.01623555  0.00500949 -0.00983441
  0.0029932  -0.00529019  0.00490932 -0.00257847  0.00653072 -0.00067597
 -0.00513989 -0.00163154  0.00285976  0.01086377  0.00819442 -0.00186532
  0.00200229 -0.00399669 -0.0104334  -0.00648835  0.00163276  0.01589233
  0.00239824  0.00094084  0.00899693 -0.00212936  0.00866897 -0.00409888
 -0.00451408  0.00946926 -0.00396359 -0.00145912  0.00438363 -0.00066124
 -0.00224987  0.01273374  0.00445319  0.00991004  0.00193672 -0.02409791
 -0.01901497  0.01049944 -0.0119889   0.01065127  0.00680365  0.00662515
  0.0030275  -0.01181102  0.01487379  0.00130858 -0

## Calculate Daily Rate of Return for SBRA

6. Call the function `simple_rate_of_return` with the arguments `adj_closings_sbra`. Then print the results. 

In [38]:
print(rate_of_return(SBRA))
srorSBRA = rate_of_return(SBRA)

[ 0.04231494  0.03223879  0.02139969  0.03227632  0.00548544 -0.00054557
 -0.00491262 -0.0093253   0.00885935  0.02469816  0.01446175  0.01319959
 -0.0020845   0.01044394  0.01498701  0.01680256  0.01552315 -0.00049301
  0.0093734   0.00391017 -0.01071081  0.01230311 -0.00631987  0.0048924
  0.00876327 -0.007722   -0.0024318  -0.01267687 -0.00148138  0.00050581
  0.01061674 -0.00800397 -0.00756427 -0.00101634 -0.00406913 -0.02655777
 -0.01364114 -0.01861692 -0.0178862  -0.02262697 -0.00395265  0.000567
 -0.00623227 -0.00171039 -0.00285556  0.01832765  0.01293593  0.01887841
  0.00163473 -0.00435251 -0.00928968  0.00220641  0.01210787  0.03371397
 -0.01578123  0.00855165  0.01695817  0.00573207  0.01709847 -0.00815086
 -0.00719044 -0.00103474 -0.00983944  0.00366116  0.01042201 -0.00464157
 -0.00362691  0.01300054  0.01026696  0.         -0.00101634 -0.02085451
 -0.0244156   0.01916937 -0.0161965   0.01646314  0.01044934  0.
  0.01292654 -0.00969886  0.00824739 -0.0015336   0.0025601   

## Calculate Daily Rate of Return for EQR

7. Call the function `simple_rate_of_return` with the arguments `adj_closings_eqr`. Then print the results. 

## Calculate Average Daily Return for SBRA

8. Use `np.mean()` with the argument `daily_simple_returns_sbra` to calculate the average daily return for SBRA. Then set it to the variable name `average_daily_simple_return_sbra`

In [39]:
mean_ror_SBRA = np.mean(srorSBRA)
mean_ror_EQR = np.mean(srorEQR)

## Calculate Average Daily Return for EQR

9. Use `np.mean()` with the argument `daily_simple_returns_eqr` to calculate the average daily return for EQR. Then set it to the variable name `average_daily_simple_return_eqr`

In [40]:
print(mean_ror_SBRA)
print(mean_ror_EQR)

0.00210937212011956
0.0015777637451981398


## Compare the Average Daily Return between EQR and SBRA

10. Based on the average daily simple returns of EQR and SBRA, which stock is more likely to be profitable in the future?

## Daily Log Returns Function

11. Create a function that returns the daily rate of return

    step 1. define a function named log_returns
    
    step 2. create parameter for  `adj_closings`
    
    step 3. use np.log() to get the log of each adjusted closing price and set it to the variable `log_adj_closings`
    
    step 4. use np.diff() to get the diff of each daily log adjusted closing price and set it to the variable `daily_log_returns`
    
    step 5. return `daily_log_returns`

In [4]:
def log_returns(adj_closings):
    log_adj_closings = np.log(adj_closings)
    daily_log_returns = np.diff(log_adj_closings)
    return daily_log_returns

## Calculate Daily Log Returns for SBRA

12. Call the function `log_returns` with the arguments `adj_closings_sbra`. Set it to the variable `daily_log_returns_sbra`. Then print the results. 

In [41]:
d_log_returns_SBRA = log_returns(SBRA)

## Calculate Daily Log Returns for EQR

13. Call the function `log_returns` with the arguments `adj_closings_eqr`. Set it to the variable `daily_log_returns_eqr`. Then print the results. 

In [42]:
d_log_returns_EQR = log_returns(EQR)

## Annualize Daily Log Return Function

14. Create a function that returns the daily rate of return

    step 1. define a function named `annualize_log_return`
    
    step 2. create parameter for  `daily_log_returns`
    
    step 3. use `np.mean()` with the argument `daily_log_returns` to calculate the average daily return. Then set it to the variable name `average_daily_log_returns`
    
    step 4. then multiply `average_daily_log_returns` by 250 and set it to the variable `annualized_log_return`
    
    step 5. return `annualized_log_return`

In [43]:
def annualized_log_return(d_log):
    avg_daily_log = np.mean(d_log)
    annual_log_ret = 250*avg_daily_log
    return annual_log_ret
    

## Calculate Annualize Daily Log Return for SBRA

10. Call the function `annualize_log_return` with the arguments `daily_log_returns_sbra`. Set it to the variable `annualized_log_return_sbra`. Then print the results. 

In [47]:
annual_log_return_SBRA = annualized_log_return(d_log_returns_SBRA)

## Calculate Annualize Daily Log Return for EQR

11. Call the function `annualize_log_return` with the arguments `daily_log_returns_eqr`. Set it to the variable `annualized_log_return_eqr`. Then print the results. 

In [48]:
annual_log_return_EQR = annualized_log_return(d_log_returns_EQR)

print(annual_log_return_SBRA)
print(annual_log_return_EQR)

0.5044563709645333
0.3855982155640554


12. Based on the differences between the Annualize Daily Log Return for EQR and SBRA, Which could be more profitable in the future and why?

## Calculate Variance of Daily Log Return for SBRA

13. Calculate the variance of the daily logarithmetic return for SBRA. Use the function `.var()` with the argument `log_daily_ror`. Set it to the variable `daily_varaince_sbra`. Then print the results. 

In [49]:
daily_var_SBRA = np.var(d_log_returns_SBRA)

## Calculate Variance of Daily Log Return for EQR

14. Calculate the variance of the daily logarithmetic return for EQR. Use the function `.var()` with the argument `daily_log_returns_eqr`. Set it to the variable `daily_varaince_eqr`. Then print the results. 

In [51]:
daily_var_EQR = np.var(d_log_returns_EQR)

print(daily_var_SBRA)
print(daily_var_EQR)

0.00017844226355047074
6.833881310511606e-05


## Compare the Variance of Daily Log Return between EQR and SBRA

15. Explain which investment is more riskier based on the Variance of daily log return between EQR and SBRA ?

## Calculate the Daily Standard Deviation for SBRA

16. Calculate the Standard Deviation of the daily logarithmetic return for SBRA. Use the function `.std()` with the argument `daily_log_returns_sbra`. Set it to the variable `daily_sd_sbra`. Then print the results. 

In [53]:
daily_sd_SBRA = np.std(d_log_returns_SBRA)

## Calculate the Daily Standard Deviation for EQR

17. Calculate the Standard Deviation of the daily logarithmetic return for EQR. Use the function `.std()` with the argument `daily_log_returns_eqr`. Set it to the variable `daily_sd_eqr`. Then print the results. 

In [54]:
daily_sd_EQR = np.std(d_log_returns_EQR)

print(daily_sd_EQR)
print(daily_sd_SBRA)

0.00826672928703463
0.013358228308816658


## Compare the Daily Standard Deviation between EQR and SBRA

18. Has your previous variance risk assessment changed based on the Daily Standard Deviation and why?

## Calculate the Correlation between SBRA and EQR

19. Calculate the Correlation of the daily logarithmetic return between SBRA and ERQ assets. Use the function `.corrcoef()` with the arguments `daily_log_returns_sbra` and `daily_log_returns_eqr`. Set it to the variable `corr_sbra_eqr`. Then print the results. 

In [57]:
corr_SBRA_EQR = np.corrcoef(d_log_returns_SBRA,d_log_returns_EQR)
print(corr_SBRA_EQR)

[[1.         0.62096591]
 [0.62096591 1.        ]]


## Interpret the Correlation between SBRA and EQR

20. Interpret and explain the correlation between the stocks SBRA and EQR?

The correlation coefficient is at 0.62 indicating a more or less strong positive correlation increasing the overall risk inside the portfolio. If economical factors are indicating flurishing of all stocks involved into real estate this risk would be acceptable. Currently it is not. 

## Final Analysis

21. Which stock would you invest in based on risk and profitability?

I would invest in SBRA due to the higher rate of return but only if i am allowed to keep the position size small enough and diversify my portfolio with non correlated stocks.
For short term investment i would prefer EQR as historical data indicates a smaller possible drawdown in the short term.